app/components/emoji_picker/emoji_picker.js

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {injectIntl, intlShape} from 'react-intl';
+import {
+    Dimensions,
+    SectionList,
+    TouchableOpacity,
+    View
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+import FormattedText from 'app/components/formatted_text';
+import SearchBar from 'app/components/search_bar';
+import {emptyFunction} from 'app/utils/general';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+const {width: deviceWidth} = Dimensions.get('window');
+const EMOJI_SIZE = 30;
+const EMOJI_GUTTER = 7.5;
+const SECTION_MARGIN = 15;
+
+class EmojiPicker extends PureComponent {
+    static propTypes = {
+        emojis: PropTypes.array.isRequired,
+        intl: intlShape.isRequired,
+        onEmojiPress: PropTypes.func,
+        theme: PropTypes.object.isRequired
+    };
+
+    static defaultProps = {
+        onEmojiPress: emptyFunction
+    };
+
+    leftButton = {
+        id: 'close-edit-post'
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            emojis: props.emojis,
+            searchTerm: ''
+        };
+    }
+
+    changeSearchTerm = (text) => {
+        this.setState({
+            searchTerm: text
+        });
+
+        clearTimeout(this.searchTermTimeout);
+        const timeout = text ? 75 : 0;
+        this.searchTermTimeout = setTimeout(() => {
+            const emojis = this.searchEmojis(text);
+            this.setState({
+                emojis
+            });
+        }, timeout);
+    };
+
+    cancelSearch = () => {
+        this.setState({
+            emojis: this.props.emojis,
+            searchTerm: ''
+        });
+    }
+
+    searchEmojis = (searchTerm) => {
+        const {emojis} = this.props;
+
+        const nextEmojis = [];
+
+        emojis.forEach((section) => {
+            const {data, ...otherProps} = section;
+            const {key, items} = data[0];
+
+            const nextData = {
+                key,
+                items: items.filter((item) => item.aliases.includes(searchTerm.toLowerCase()))
```

Comment:
maybe we can move the `searchTerm.toLowerCase()` up so it only is executed once and not for every filter?